2019.07

montjoie opened the pull request.

montjoie committed:
Upgrade to 2019.07

This patch upgrade LAVA to 2019.07 via their official docker images.
Along with the change of the baseimage from our lava-xx-base to official
2019.07, some minor changes are needed:
- Activate the en_US.UTF-8 locale needed for postgresql
- chown to lavaserver all copied device-types
- Fix the start scripts for using the official entrypoints

montjoie committed:
Add support for device-type alias

This patch adds support for the device-type aliases.

khilman commented on lava-master/Dockerfile:
```diff
@@ -1,4 +1,7 @@
+FROM lavasoftware/lava-server:2019.07
+
+RUN apt-get update
+RUN apt-get -y install lava-dispatcher sudo
```
Don't you need the `apt-get update` and `apt-get install` on the same line to avoid docker caching issues?

montjoie replied:
OK

khilman commented on lava-master/Dockerfile:
```diff
@@ -36,6 +39,9 @@ COPY env/ /etc/lava-server/dispatcher.d/
 
 COPY apache2/ /etc/apache2/
 
+# Thoses two line fixes 'postgresql ERROR:  invalid locale name: "en_US.UTF-8"' when restoring a backup
```
which two lines?  I only see one line added?

montjoie replied:
I compressed thoses two lines in one, I update the comment

khilman commented on lava-master/Dockerfile:
```diff
@@ -1,4 +1,6 @@
-FROM baylibre/lava-master-base:2019.03_stretch
+FROM lavasoftware/lava-server:2019.07
+
+RUN apt-get update && apt-get -y install lava-dispatcher sudo lava-coordinator
```
Doesn't `apt-get update` need a `-y`?
Why is lava-dispatcher needed in the lava-master image?

montjoie replied:
apt-get update does not need -y since it didnt ask question.
I have removed lava-dispatcher

khilman commented on lava-slave/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM baylibre/lava-slave-base:2019.03_stretch
+FROM lavasoftware/lava-dispatcher:2019.07
```
Now that we switch to the official images, this patch sould also remove the lava-master-base and lava-slave-base directories.

montjoie replied:
Removed!

montjoie committed:
Use yaml safe_load insteald of load

On my gentoo, using yaml.load now give:
  Traceback (most recent call last):
  raise RuntimeError("Unsafe load() call disabled by Gentoo. See bug #659348")
  RuntimeError: Unsafe load() call disabled by Gentoo. See bug #659348

Note that on recent ubuntu, a warning appears also.
  YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.

This is due to a security risk of using yaml.load()

Since we didnt rely on any behavour provided by load(), let's convert the call to safe_load().
